pkg/controllers/scheduler/util.go

```diff
@@ -98,3 +100,93 @@ func getResourceRequest(
 	resource := resourceutil.GetPodResourceRequests(podSpec)
 	return *framework.NewResource(resource), nil
 }
+
+func getCustomMigrationInfoBytes(existingMigrationConfigBytes string,
+	fedObj fedcorev1a1.GenericFederatedObject, ftc *fedcorev1a1.FederatedTypeConfig,
+) (string, error) {
+	existingMigrationConfig := &framework.MigrationConfig{}
+	err := json.Unmarshal([]byte(existingMigrationConfigBytes), existingMigrationConfig)
+	if err != nil {
+		return "", err
+	}
+
+	err = validateMigrationConfig(existingMigrationConfig, fedObj, ftc)
+	if err != nil {
+		return "", err
+	}
+
+	customMigrationInfo := convertToCustomMigrationInfo(existingMigrationConfig)
+	customMigrationInfoBytes, err := json.Marshal(customMigrationInfo)
+	if err != nil {
+		return "", err
+	}
+
+	return string(customMigrationInfoBytes), nil
+}
+
+func validateMigrationConfig(migrationConfig *framework.MigrationConfig, fedObj fedcorev1a1.GenericFederatedObject,
+	ftc *fedcorev1a1.FederatedTypeConfig,
+) error {
+	clusters := sets.New[string]()
+
+	for _, replicasMigration := range migrationConfig.ReplicasMigrations {
+		if replicasMigration.ClusterName == "" {
+			return fmt.Errorf("clusterName cannot be empty")
+		}
+		if replicasMigration.LimitedCapacity == nil {
+			return fmt.Errorf("limitedCapacity cannot be empty")
+		}
+		if *replicasMigration.LimitedCapacity < 0 {
+			return fmt.Errorf("limitedCapacity cannot be less than zero")
+		}
+
+		sourceObj, err := fedObj.GetSpec().GetTemplateAsUnstructured()
```

Comment:
This should be done outside the for loop?

Reply:
Agreed.

Reply:
nit: we don't need to unmarshal the template if `len(migrationConfig.ReplicasMigration) == 0`